test/quantization/test_quantized_op.py

```diff
@@ -502,32 +502,89 @@ def test_qthreshold(self, X, threshold, value):
             qY = op(qX, threshold, value)
             self.assertEqual(qY, qY_hat, msg="{} qthreshold failed".format(name))
 
-    """Tests the correctness of the quantized::clamp op."""
+    """Tests the correctness of the quantized torch.clamp op."""
     @given(X=hu.tensor(shapes=hu.array_shapes(1, 8, 1, 8, max_numel=10**5),
                        elements=hu.floats(-1e6, 1e6, allow_nan=False),
                        qparams=hu.qparams()),
            min_val=hu.floats(-1e6, 1e6, allow_nan=False),
            max_val=hu.floats(-1e6, 1e6, allow_nan=False))
     def test_qclamp(self, X, min_val, max_val):
         X, (scale, zero_point, torch_type) = X
+        X = torch.from_numpy(X)
+
+        def quantize_with_same_params(tensor):
+            return torch.quantize_per_tensor(tensor, scale=scale,
+                                             zero_point=zero_point,
+                                             dtype=torch_type)
+
+        # Create min/max argument product
+        min_tensors = (torch.rand_like(X) * 2e5 - 1e5,  # [-1e5, 1e6)
+                       torch.rand_like(X[0]) * 2e5 - 1e5,  # Broadcasting
+                       min_val,  # With scalars
+                       )
+        max_tensors = (torch.rand_like(X) * 2e5 - 1e5,  # [-1e5, 1e6)
+                       torch.rand_like(X[-1]) * 2e5 - 1e5,   # Broadcasting
+                       max_val  # With scalars
+                       )
+        args = itertools.product(min_tensors, max_tensors)
+
+        qX = quantize_with_same_params(X)
+
+        for min_tensor, max_tensor in args:
+            # Quantize tensors
+            if isinstance(min_tensor, torch.Tensor):
+                q_min = quantize_with_same_params(min_tensor)
+            else:
+                q_min = min_tensor
+            if isinstance(max_tensor, torch.Tensor):
+                q_max = quantize_with_same_params(max_tensor)
+            else:
+                q_max = max_tensor
+
+            # Use floating clamp implementation as reference
+            Y = torch.clamp(X, min_tensor, max_tensor)
+            qY = quantize_with_same_params(Y)
+            qY_hat = torch.clamp(qX, q_min, q_max)
+            self.assertEqual(qY, qY_hat, msg="quantized torch.clamp failed")
+
+    """Tests the correctness of the quantized torch.clamp op when argument have different qparams"""
+    @given(X=hu.tensor(shapes=hu.array_shapes(1, 8, 1, 8),
+                       elements=hu.floats(-1e6, 1e6, allow_nan=False),
+                       qparams=hu.qparams()),
+           qparams=hu.qparams())
+    def test_qclamp_with_tensors_different_quant_params(self, X, qparams):
+        X, (scale, zero_point, torch_type) = X
+        other_scale, other_zero_point, other_dtype = qparams
+
+        # Make sure that the other qparams are different
+        assume(other_dtype != torch_type
+               or other_zero_point != zero_point 
+               or other_scale != scale)
```

Comment:
if we remove this check, the single test case would work for checking all the logic